crates/backend/src/codegen.rs

```diff
@@ -984,34 +982,74 @@ impl ToTokens for ast::ImportType {
             }
         };
 
+        let generic_defs;
+        let generics;
+        let phantom;
+        let phantom_init;
+        let default_as_ref;
+
+        let where_clause = generics::where_clause(&self.generics);
```

Comment:
These custom utils seem to duplicate what [`syn::Generics::split_for_impl`](https://docs.rs/syn/latest/syn/struct.Generics.html#method.split_for_impl) is doing.

You might want to use it instead for simplicity.

Reply:
Thanks, will look into this.

Reply:
I've added using `split_for_impl` here instead.